cmd/agent/subcommands/snmp/command.go

```diff
@@ -107,8 +109,43 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 					port = defaultPort
 				}
 			} else {
+				//If the customer provides only 1 argument : the ip_address
+				//We check the ip address configuration in the agent runtime and we use it for the snmpwalk
 				deviceIP = address
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** If have a command `datadog-agent snmp walk -C comString1 1.2.3.4` and `1.2.3.4` is also present in conf.yaml with:

```
instances:
  - ip_address: 1.2.3.4
    community_string: comString2
```

In this case, does it mean, the command will ignore the argument ` -C comString1` and use `comString2` ? If so, I think this might be confusing for users.

It seems that merging configs from command line and configs from runtime snmp instances configs might create confusions.

Maybe we can: 
- 1/ if either community string (v2) or user (v3) is provided, then do not use config from runtime snmp instances, but the issue is that, by default, if -C is not provided, it already means we should use `public`.
- 2/ use a new flag to express that we want to use runtime configs
- 3/ use a separate sub command

Also, in case we want to merge config from runtime instances and command line configs, the command line configs should take precedence (this seems not the case right now). For example, if IP is 1.2.3.4, AuthKey is set in both runtime instance config and as command line argument.

‏

Reply:
I think it is cooler to use the same command, by default looking for the config and loading params from there, overriding them from the command line if explicitly configured. 

It is pretty common, we might be more explicit about using agent configuration in the helper. 

Note this is an snmp walk internal to the agent: the goal is not to be closer to snmp walk, but rather to the agent. So no surprise to me that we load by default the configuration, super duper useful to help debugging customers !

Reply:
I'm thinking about the case where we have different values for `context` which as we saw in some examples before would give different outputs according to the value, if the user gives a certain value in the config but still wants to run a walk with a second `context` then using this logic would give the false result.